python/cudf_polars/cudf_polars/experimental/select.py

```diff
@@ -27,10 +58,90 @@ def _(
     if pi.count > 1 and not all(
         expr.is_pointwise for expr in traversal([e.value for e in ir.exprs])
     ):
-        # TODO: Handle non-pointwise expressions.
-        raise NotImplementedError(
-            f"Selection {ir} does not support multiple partitions."
-        )
+        try:
+            # Try decomposing the underlying expressions
+            return decompose_select(ir, child, partition_info)
+        except NotImplementedError:
+            # TODO: Handle more non-pointwise expressions.
+            return _lower_ir_fallback(
+                ir, rec, msg="This selection is not supported for multiple partitions."
+            )
+
     new_node = ir.reconstruct([child])
     partition_info[new_node] = pi
     return new_node, partition_info
+
+
+def build_fusedexpr_select_graph(
+    ir: Select, partition_info: MutableMapping[IR, PartitionInfo]
+) -> MutableMapping[Any, Any]:
+    """Build complex Select graph."""
+    (child,) = ir.children
+    child_count = partition_info[child].count
```

Comment:
*nit*: Naming, I think that this is not really a "child" once we're processing each expression.